src/webxdc.rs

```diff
@@ -365,35 +364,97 @@ impl Context {
             .await?;
 
         if send_now {
-            // send update now
-            // (also send updates on MessagesState::Failed, maybe only one member cannot receive)
-            let mut status_update = Message {
-                chat_id: instance.chat_id,
-                viewtype: Viewtype::Text,
-                text: Some(descr.to_string()),
-                hidden: true,
-                ..Default::default()
+            let guard = self.status_updates_mutex.lock().await;
+            match self
+                .sql
+                .query_get_value(
+                    "SELECT first_serial FROM smtp_status_updates WHERE msg_id=?",
+                    paramsv![instance.id],
+                )
+                .await?
+            {
+                Some(first_serial) => {
+                    if status_update_serial <= first_serial {
+                        bail!("bad serial");
+                    }
+                    self.sql
+                        .execute(
+                            "UPDATE smtp_status_updates SET last_serial=?, descr=? WHERE msg_id=?",
```

Comment:
this is a clever sql statement. however, wondering if the mutex is still needed because of SELECT/DELETE needs to be executed together. or we would need to capsule that into an `IMMEDIATE TRANSACTION`

Reply:
there seems to be sth. as `DELETE ... RETURNING`, https://www.sqlite.org/lang_returning.html, that could do the job. it is pretty new in sqlite, available since 3.35.0 (2021-03-12), however, iirc, we ship our own sqlite, so this should not be an issue.

Reply:
i added commits to use `DELETE ... RETURNING` and removing the mutex.